docs/source/basics/instruct_datasets.rst

```diff
@@ -0,0 +1,233 @@
+.. _instruct_dataset_usage_label:
+
+=================
+Instruct Datasets
+=================
+
+Instruction tuning involves training an LLM to perform specific task(s). This typically takes the form
+of a user command or prompt and the assistant's response, along with an optional system prompt that
+describes the task at hand. This is more structured than freeform text association that models are
+typically pre-trained with, where they learn to specifically predict the next token instead of completing
+the task.
+
+The primary entry point for fine-tuning with instruct datasets in torchtune is the :func:`~torchtune.datasets.instruct_dataset`
+builder. This lets you specify a local or Hugging Face dataset that follows the instruct data format
+directly from the config and train your LLM on it.
+
+Instruct dataset format
+-----------------------
+
+Instruct datasets typically follow the input-output format, where the user prompt is in one column
+and the assistant prompt is in another column.
+
+.. code-block:: text
+
+    |  input          |  output          |
+    |-----------------|------------------|
+    | "user prompt"   | "model response" |
+
+The user input column is converted to a :class:`~torchtune.data.Message` with ``role="user"`` and the assistant
+output column is converted to a a :class:`~torchtune.data.Message` with ``role="assistant"``. These are then pass into
+the model tokenizer's ``tokenize_messages`` to encode and add appropriate model-specific special tokens.
```

Comment:
I think this part could be clearer. Why not just give an explicit example of each stage? Like you have the raw data format already, which is great. Then show how the same example would look in the Message representation, and finally the format after the messages have been tokenized. But as a first-time reader I see "tokenize_messages" and "model-specific special tokens" and I'm like wth is this

Reply:
no no to goal is to obscure further